src/Sylius/Component/Order/Modifier/OrderItemQuantityModifier.php

```diff
@@ -42,6 +42,8 @@ public function modify(OrderItemInterface $orderItem, $targetQuantity)
             return;
         }
 
+        $targetQuantity = min($targetQuantity, 9999);
```

Comment:
Anyway, we can implement this behaviour as a decorator for `OrderItemQuantityModifierInterface`.